libqtile/widget/groupbox.py

```diff
@@ -50,17 +50,26 @@ def __init__(self, **config):
         self.add_defaults(base.PaddingMixin.defaults)
         self.add_defaults(base.MarginMixin.defaults)
 
+        if "borderwidth" in config:
+            logger.warning(
```

Comment:
Make sure `logger` is imported:
``` python
from libqtile.log_utils import logger
```
You may need to add this to the other files you changed.